### PR TITLE
 Fixing #3869 

### PR DESCRIPTION
**Pull request by mushthaq33**

### CODE REVIEW

**Comment by bhathiya** [comment]
What's the issue we fix here? Can't find a link.

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java [hunk 1]
```diff
@@ -344,14 +345,35 @@ public static NativeObject jsFunction_getApplicationKey(Context cx, Scriptable t
...
+                    appInfoMap.put("allowedDomains", accessAllowDomainsArray);
+                    appInfoMap.put("validityPeriod", validityPeriod);
+                    appInfoMap.put("scopes", scopes);
+                    appInfoMap.put("groupingId", groupingId);
+                    appInfoMap.put("jsonParams", jsonParams);
+                    appInfoMap.put("applicationId", applicationId);
+                    keyDetails = getAPIConsumer(thisObj).requestApprovalForApplicationRegistrationByApplicationId(appInfoMap);
+                } else {
+                    keyDetails = getAPIConsumer(thisObj).requestApprovalForApplicationRegistration(
+                            username, applicationName, tokenType, callbackUrl,
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: i think we can merge this in to two lines.

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java [hunk 2]
```diff
@@ -2543,6 +2586,40 @@ public static SubscriptionResponse jsFunction_addAPISubscription(Context cx, Scr
         return addSubscriptionResponse;
     }
 
+    public static SubscriptionResponse jsFunction_addAPISubscriptionByAppId(Context cx, Scriptable thisObj,
+            Object[] args, Function funObj)
+            throws APIManagementException {
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: merge lines 2590 and 2591

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 3]
```diff
@@ -3200,6 +3415,58 @@ else if (APIConstants.API_KEY_TYPE_SANDBOX.equals(tokenType)) { // if
         return keyDetails;
     }
 
+    @Override
+    public Map<String, String> completeApplicationRegistration(String userId, int applicationId, String tokenType, String tokenScope, String groupingId) throws APIManagementException {
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Is this a newly added method?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 4]
```diff
@@ -3700,6 +3972,60 @@ public OAuthApplicationInfo updateAuthClient(String userId, String applicationNa
...
+     * @param validityTime validity time period.
+     * @param groupingId APIM application id.
+     * @param jsonString Callback URL for the Application.
+     * @param tokenScope Scopes for the requested tokens.
+     * @return
+     * @throws APIManagementException
+     */
+    @Override
+    public OAuthApplicationInfo updateAuthClientByAppId(String userId, String applicationName, int applicationId,
+            String tokenType,
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: do we need separate lines for each variable?

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/application/application-update/ajax/application-update.jag [hunk 5]
```diff
@@ -74,6 +74,74 @@ include("/jagg/jagg.jag");
...
+        print(obj);
+    } else if (action == "updateApplicationById" && request.getMethod() == 'POST') {
+        if (!user) {
+            if (!ssoEnabled) {
+      		    print({
+      		        error:true,
+                    message:msg.error.loginRequired(action)
+            });
+      	    }
+      	    return;
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/application/application-update/ajax/application-update.jag]: formatting correct?

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-add/ajax/subscription-add.jag [hunk 6]
```diff
@@ -108,7 +109,41 @@ include("/jagg/jagg.jag");
...
         print(obj);
-    }else if (action == "generateAPIKey" && request.getMethod() == 'POST') {
+    } else if (action == "addAPISubscriptionByAppId" && request.getMethod() == 'POST') {
+        if (!user) {
+            if (!ssoEnabled) {
+      		    print({
+      		        error:true,
+                    message:msg.error.loginRequired(action)
+                });
+      	    }
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-add/ajax/subscription-add.jag]: pls check formatting

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/js/applications.js [hunk 7]
```diff
@@ -630,8 +637,8 @@ $("#application-actions").each(function(){
         $('#messageModal a.btn-other').html(i18n.t("No"));
         $('#messageModal a.btn-primary').click(function() {
             jagg.post("/site/blocks/application/application-remove/ajax/application-remove.jag", {
-                action:"removeApplication",
-                application:appName
+                action:"removeApplicationById",
+                applicationId:appId
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/js/applications.js]: to make it consistent shall we change the param applicationId to appId?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java (hunk 3 above)
Comment (mushthaq33) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Yes. This is a newly added method to support application registration using application ID

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java (hunk 1 above)
Comment (mushthaq33) [reply components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Fixed with 89b6956b5c1641bd96ee670c827bbf6d0a87550a

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java (hunk 2 above)
Comment (mushthaq33) [reply components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Fixed with 89b6956b5c1641bd96ee670c827bbf6d0a87550a

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java (hunk 4 above)
Comment (mushthaq33) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Fixed with 89b6956b5c1641bd96ee670c827bbf6d0a87550a

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/application/application-update/ajax/application-update.jag (hunk 5 above)
Comment (mushthaq33) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/application/application-update/ajax/application-update.jag]: Fixed with 4446c3df369e717d3f6eb1bbbda3726792114ad3

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-add/ajax/subscription-add.jag (hunk 6 above)
Comment (mushthaq33) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-add/ajax/subscription-add.jag]: Fixed with 4446c3df369e717d3f6eb1bbbda3726792114ad3

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/js/applications.js (hunk 7 above)
Comment (mushthaq33) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/js/applications.js]: Fixed with 89b6956

**Pull request merged**